component/org.wso2.carbon.identity.oauth2.grant.token.exchange/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/utils/TokenExchangeUtils.java

```diff
@@ -749,4 +856,71 @@ private static boolean validateUsingCertificate(SignedJWT signedJWT, IdentityPro
     }
+
+    public static User getLocalUser(OAuthTokenReqMessageContext tokReqMsgCtx,
+                                    String subjectIdentifier) throws IdentityOAuth2Exception {
+
+        User localUser = null;
+        AbstractUserStoreManager userStoreManager = getUserStoreManager(tokReqMsgCtx);
+        try {
+             if(userStoreManager.isExistingUser(subjectIdentifier)) {
+                 localUser = userStoreManager.getUser(null, subjectIdentifier);
+             }
+        } catch (UserStoreException e) {
+            handleException("Error while resolving local user for subject: " + subjectIdentifier);
+        }
+        return localUser;
+    }
+
+    private static String resolveSubjectIdentifier(JWTClaimsSet claimsSet, String[] lookupAttributes)
+            throws IdentityOAuth2Exception {
+
+        String subjectIdentifier = null;
+        for (String lookupAttribute : lookupAttributes) {
+            Object subjectIdentifierObj = claimsSet.getClaim(lookupAttribute);
+            if (subjectIdentifierObj instanceof String) {
+                subjectIdentifier = (String) subjectIdentifierObj;
+            }
+
+            if (lookupAttribute.equals("email") && StringUtils.isBlank(subjectIdentifier)) {
+                String regex = "^(.+)@(.+)$";
+                Pattern pattern = Pattern.compile(regex);
+                Matcher matcher = pattern.matcher(claimsSet.getSubject());
+                if (matcher.matches()) {
+                    subjectIdentifier = claimsSet.getSubject();
+                }
+            }
+        }
+
+        if (StringUtils.isBlank(subjectIdentifier)) {
+            throw new IdentityOAuth2Exception("Required claim not found in the token");
+        }
+        return subjectIdentifier;
+    }
+
+    public static AbstractUserStoreManager getUserStoreManager(OAuthTokenReqMessageContext tokReqMsgCtx)
+            throws IdentityOAuth2Exception {
+
+        RealmService realmService = TokenExchangeComponentServiceHolder.getInstance().getRealmService();
+        String tenantDomain = tokReqMsgCtx.getOauth2AccessTokenReqDTO().getTenantDomain();
+        AbstractUserStoreManager userStoreManager = null;
+
+        if (StringUtils.isEmpty(tenantDomain)) {
+            tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+        }
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+
+        try {
+            UserRealm realm = (UserRealm) realmService.getTenantUserRealm(tenantId);
+
+            if (realm.getUserStoreManager().getSecondaryUserStoreManager() != null) {
+                userStoreManager = (AbstractUserStoreManager) realm.getUserStoreManager().getSecondaryUserStoreManager();
+            } else {
+                userStoreManager = (AbstractUserStoreManager) realm.getUserStoreManager();
+            }
+        } catch (UserStoreException e) {
+            handleException("Error while getting user store manager: " + e.getMessage());
```

Comment:
'e' is ignored

Reply:
addressed with [4f8ccec](https://github.com/wso2-extensions/identity-oauth2-grant-token-exchange/pull/19/commits/4f8ccecff35a7fd0c1a8486fb8f56a12e1f2b8c7)